### PR TITLE
Merge MVC into Development

### PR DESCRIPTION
**Pull request by InclementDab**

### CODE REVIEW

File: JM/CF/Scripts/3_Game/CommunityFramework/MVC/Controller.c [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+class Controller: MVCEventHandler
+{
+	// Private members
+	private ref PropertyHashMap m_ControllerPropertyHashMap;
+	
+		
+	protected ref DataBindingHashMap m_ViewBindingHashMap = new DataBindingHashMap();
+	protected ref PropertyHashMap m_PropertyHashMap = PropertyHashMap.FromType(Type());
+	typename GetPropertyType(string property_name) {
```
Comment (Jacob-Mango) [inline JM/CF/Scripts/3_Game/CommunityFramework/MVC/Controller.c]: Please follow code formatting

File: JM/CF/Scripts/3_Game/CommunityFramework/MVC/Controller.c [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+		return m_PropertyHashMap.Get(property_name);
+	}
+	
+	
+	
+	
+	void Controller()
+	{
+		GetLogger().Log("Controller", "JM_CF_MVC");
```
Comment (Jacob-Mango) [inline JM/CF/Scripts/3_Game/CommunityFramework/MVC/Controller.c]: Don't use this logger

File: JM/CF/Scripts/3_Game/CommunityFramework/MVC/Controller.c [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+		
+		return binding_map.Count();
+	}
+	
+	
+	
+}
+
+
+
```
Comment (Jacob-Mango) [inline JM/CF/Scripts/3_Game/CommunityFramework/MVC/Controller.c:245]: So many new lines. Please keep the code clean!

**Review by Jacob-Mango — CHANGES_REQUESTED** [review]

File: JM/CF/Scripts/3_Game/CommunityFramework/MVC/Controller.c (hunk 1 above)
Comment (InclementDab) [reply JM/CF/Scripts/3_Game/CommunityFramework/MVC/Controller.c]: Only Getters and Setters use 1-line bracket formatting. Keeps things much cleaner and makes them easier to differentiate from proper methods

File: JM/CF/Scripts/3_Game/CommunityFramework/MVC/Controller.c (hunk 2 above)
Comment (InclementDab) [reply JM/CF/Scripts/3_Game/CommunityFramework/MVC/Controller.c]: Fixed

File: JM/CF/Scripts/3_Game/CommunityFramework/MVC/Controller.c (hunk 3 above)
Comment (InclementDab) [reply JM/CF/Scripts/3_Game/CommunityFramework/MVC/Controller.c:245]: _**bruh**_

**Review by Jacob-Mango — APPROVED** [review]

**Review by Arkensor — APPROVED** [review]

**Pull request merged**